internal/utils/test/mock/terminal_ui.go

```diff
@@ -88,6 +89,26 @@ func NewConsoleWithOptions(options UIOptions, writers ...io.Writer) (*expect.Con
 		console.Tty(),
 		console.Tty(),
 		console.Tty(),
+		terminal.NewUIBrowser(),
+	)}
+
+	return console, ui, nil
+}
+
+// NewVT10XConsoleWithBrowser returns a *expect.Console and mock terminal UI that write to the buffer
+func NewVT10XConsoleWithBrowser(browser terminal.Browser) (*expect.Console, terminal.UI, error) {
```

Comment:
I can't comment on it, but if you don't want to have to add more helpers to this file for "with mocked browser" then I might suggest adding this to `UIOptions`:

type UIOptions struct {
	AutoConfirm bool
	UseColors   bool
	UseJSON     bool
	OpenBrowserFn     func(url string) error
}
```

and then just setting that on the returned `ui` value from each of these helpers